src/comm/Config.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const commUtils = require('./util');
+
+const nconf = require('nconf');
+nconf.formats.yaml = require('nconf-yaml');
+
+//
+// The class representing the hierarchy of configuration settings.
+//
+const Config = class {
+    /**
+     * Constructor
+     */
+    constructor() {
+        nconf.use('memory');
+        nconf.use('mapenv', {type:'memory'});
+        this.mapSettings(nconf.stores.mapenv, process.env);
+        this._fileStores = [];
+        // reference to configuration settings
+        this._config = nconf;
+        const defaultConfig = commUtils.resolvePath('config/default.yaml');
+        this.file(defaultConfig);
```

Comment:
This should be moved after the `nconf.env()` line to make it the lowest priority.

Reply:
Fixed